insights/parsers/pci_rport_target_disk_paths.py

```diff
@@ -0,0 +1,219 @@
+        """
+        return self.__attrs__('host')
+
+    @property
+    def rport(self):
+        """
+        The all rport(s) from parsed content.
+
+        Returns:
+            list: rports
+        """
+        return self.__attrs__('rport')
+
+    @property
+    def target(self):
+        """
+        The all rport(s) from parsed content.
+
+        Returns:
+            list: rports
+        """
+        return self.__attrs__('target')
+
+    @property
+    def pci(self):
+        """
+        (dict): The full information parsed
+        """
+        return self
+
+    @property
+    def host_channel_id_lun(self):
+        """
+        The all host_channel_id_lun(s) from parsed content
+
+        Returns:
+            list: host_channel_id_lun
+        """
+        return self.__attrs__('host_channel_id_lun')
+
+    def parse_content(self, content):
+        EMPTY = "Input content is empty"
+        BADWD = "No useful data parsed in content: '{0}'".format(content)
+
+        if not content:
+            raise SkipException(EMPTY)
+
+        pdata_list = []
+        pci = {'PCI': pdata_list}
+
+        for line in content:
+            line = line.strip()
+
+            temp_pci = {}
+            cnt = 0
+
+            host_pos = None
+            pci_id_pos = None
+            rport_pos = None
+            target_pos = None
+            devnode_pos = None
+            host_channel_id_lun = None
+
+            for l in line.split('/', 11):
+                if 'host' in l:
+                    host_pos = cnt
+                if 'pci' in l:
+                    pci_id_pos = cnt + 2
+                if 'rport-' in l:
+                    rport_pos = cnt
+                if 'target' in l:
+                    target_pos = cnt
+                if 'block' == l:
+                    devnode_pos = cnt + 1
+                    host_channel_id_lun_pos = cnt - 1
+                cnt += 1
+
+            if (host_pos and pci_id_pos and target_pos and devnode_pos and host_channel_id_lun_pos):
+                pci_id = line.split('/', 11)[pci_id_pos]
+                host = line.split('/', 11)[host_pos]
+                target = line.split('/', 11)[target_pos]
+                devnode = line.split('/', 11)[devnode_pos]
+                host_channel_id_lun = line.split('/', 11)[host_channel_id_lun_pos]
+                if rport_pos:
+                    rport = line.split('/', 11)[rport_pos]
+                    temp_pci['rport'] = rport
+                else:
+                    temp_pci['rport'] = None
+
+                temp_pci['pci_id'] = pci_id
+                temp_pci['host'] = host
+                temp_pci['target'] = target
+                temp_pci['devnode'] = devnode
+                temp_pci['host_channel_id_lun'] = host_channel_id_lun
+
+                if temp_pci not in pdata_list:
+                    pdata_list.append(temp_pci)
+            else:
+                raise ParseException(BADWD)
+        self.update(pci)
```

Comment:
@xhe123 by adding all of the properties to this class it does not really look like a `dict` object anymore.  I think it would be better to just base the class on `CommandParser` and keep `pci` as an attribute of the class.  Since `pci` is a dictionary it would provide all of the same functionality that you are creating here.

Reply:
@bfahr yes, I agree with you. I leave the property of pci, removed others as you suggested.

Moreover, considering much about I can invoke these functions frequently in the coming insights rule, I keep these class functions so that I am able to get these item list directly through obj.pci_id()[0], alternative , users can also use them like this: obj.pci['PCI'][0]['pci_id'] . Does it look ok to you?